src/message.coffee

```diff
@@ -57,45 +62,96 @@ class SlackTextMessage extends TextMessage
       attachment_text = @rawMessage.attachments.map(a => a.fallback).join('\n')
       text = text + '\n' + attachment_text
 
-    # replace links
-    text = text.replace SlackTextMessage.MESSAGE_REGEX, (m, type, link, label) =>
+    # Replace links in text async to fetch user and channel info (if present)
+    @replaceLinks(text).then((replacedText) =>
+
+      text = replacedText
+      text = text.replace /&lt;/g, '<'
+      text = text.replace /&gt;/g, '>'
+      text = text.replace /&amp;/g, '&'
+
+      if @_channel?.is_im
+        text = "#{@_robot_name} #{text}"     # If this is a DM, pretend it was addressed to us
+
+      @text = text
+
+      cb()
+    )
+
+  ###*
+  # Replace links inside of text
+  ###
+  replaceLinks: (text) ->
+    parts = []
+    cursor = 0
+    regex = SlackTextMessage.MESSAGE_REGEX
+    regex.lastIndex = 0
+
+    while (res = regex.exec(text))
```

Comment:
what happened to the [string-replace-async](https://www.npmjs.com/package/string-replace-async) module? this loop looks a little complex.